### DISCUSSION TITLE
8-0 Part 3 - Georeferencing Web Tiles WORKAROUND

### DISCUSSION THREAD

**Discussion by kev-bone** [ua-gist604b-f24/syllabus#42] [General]
I thought I'd share how I figured out how to access the Stamen map tiles, to save others the trouble (and keep myself from online shopping). In this part of the exercise we are using an XML file with a URL to access the Stamen terrain background. If you try and run it as is you will get an error with the URL that it could not resolve the host.

It turns out Stamen map tiles are now served by Stadia Maps. Unfortunately this means they are no longer free! But you can start a 14 day trial with Stadia Maps (no credit card info needed). I followed the steps on this: [Migration Guide](https://docs.stadiamaps.com/guides/migrating-from-stamen-map-tiles/)

I created a Stadia Maps account, then on the dashboard went to Create a Property (give it whatever name), and that will generate your API key. Back on the Migration Guide, use the Stadia URL format of the Terrain style under URL-Based Migration Steps. Take this link, append your API key to it, and then replace the ServerURL link in the XML file. Keep the $ signs in front of z,x,y. It should end up like:

`<ServerUrl>https://tiles.stadiamaps.com/tiles/stamen_terrain/${z}/${x}/${y}.jpg?api_key=YOUR-API-KEY</ServerUrl>`

Hopefully that works! Just replace 'terrain' with 'watercolor' in the URL when repeating the steps.

**Comment by travisz09** [comment]
Just to add to this, Stadia Maps also have a free subscription tier, with a limited number of monthly credits. I've linked my account to both my ArcPro and QGIS, and I've never reached the limit of the free tier.

**Reply by aaryno** [reply]
Thank you for sharing this, Travis.

**Comment by jenw-ua** [comment]
Thanks Kevin. This worked for me! Did you have any trouble submitting a pull request for this assignment? All of my files are names correctly but it's acting like they're not in my submission and giving me errors because they're "missing"...yet, they're there. Strange. I ran it again and then I it accepted a few of them but it's not recognizing the others.

**Reply by kev-bone** [reply]
Yes, same! Perhaps the Check Outputs was not set up with the exact names. But I would think all the files being uploaded to your branch should be okay.

**Reply by jenw-ua** [reply]
I was thinking there was just a typo somewhere because it looks like there's a space after the names. Not sure if that's normal. So just upload and leave it, should be good, right?
<img width="668" alt="Screenshot 2024-12-03 at 12 47 13 PM" src="https://github.com/user-attachments/assets/c4193176-0650-4fda-a7c1-123e41ee42f6">

**Comment by wtb4rtr** [comment]
HI I just submitted my pull request for this module and I also failed the check for the exact reason listed above; did anyone figure out why this is? Thanks
Bill Brown

**Reply by vmwielgus** [reply]
I also failed the check. It said all my required files were missing which is strange.

**Comment by wtb4rtr** [comment]
PS Thank you for the tip Kevin; I got everything to work because of your help (except my pull request failed, but I got the required maps)
Bill Brown

**Comment by aldunn4** [comment]
Thanks for this help. My pull request also had checks that failed, but I have all the files.  🤷‍♀️

**Comment by vmwielgus** [comment]
As others have said thanks Kevin for your help!

**Comment by khartman-nj** [comment]
![image](https://github.com/user-attachments/assets/6ba97e77-4a83-46f3-85fa-7085b5268570)
aby idea why i am getting a bad token request?

**Reply by kev-bone** [reply]
Open up the XML file (orange wifi symbol) and replace the old Server URL with the new one with your API key. Then start from the first gdal_translate in this section.
